File: REply/REply/q3/BarberShop.cpp
#include<Barber.cpp>
class Customer {
}

class BarberShop
{
  
  private:

  private Customer customer;
  private boolean isNewCustomer;  // Private members , object of customer class and check if isNewCustomer is there.
  

  public:
    Queue<Customer> waitingcustomers; //Queue for waiting customers, size of queue is equal to no of chairs.
    Customer getcustomer()
  {
    return this.customer;  //getter for customer
  }
   
   void setNewCustomer(Customer customer)
  {
    this.customer = customer;  //Setter for customer
  }



}

int main()
{

    BarberShop barbshop = new BarberShop();
    Barber barber = new Barber();  //object of barber class
    cin>> no_of_chairs;  //input no of chairs in barber shop

    while(1)
    {
        if(!barber.issleeping())
        {
            if(barbshop.waitingcustomers.size() > 0)
                {
                    barber.workWithCustomer(barbshop.waitingcustomers.pop());     //if barber is not sleeping, assign it a waiting customer in the queue.
                    if(barbshop.waitingcustomers.size() == 0)                     // if queue becomes empty after waiting customer is processed, assign it to sleeping state.
                        barber.setsleeping(true);

                }

        }

        if(barbshop.isNewCustomer)
        {
             Customer c = barbshop.getNewCustomer();
                if (barber.issleeping())
                {
                    barber.awake();  // when new customer arises, and barber is sleeping, wake up the barber.
                }
                 if(barber.isworking())
                 {
                    if(barbshop.waitingcustomers.size() < nof_of_chairs)    //if barber is working, put new customer in the waiting queue.
                        waitingcustomers.push(c);
                    else
                        cout<<"customer goes to another shop";        // if size of waiting queue become equal to no of avavilable chairs, barber goes to new shop
                 } 
                
        }
         
        
    }

}

Hi Paul,

I have started my thesis at ETH, Zurich on ''. 

Currently i am doing the literature survey, 
1. Checking what else have been by app market analysis data. 
2. What are the new emerging applications

I have also collected data from 42mattersAG, 



Thanks and Regards
Priya Goel
Student MSc Informatics(03668914)
Technical University Munich

File: REply/REply/q3/BarberShop.java
package q3;
import java.util.LinkedList;
import java.util.Queue;

/**
 * Created by ramakantagrawal on 11/02/17.
 */
public class BarberShop {


    private Boolean isNewCustomer;
    private Customer customer;
    public Queue<Customer> chairs = new LinkedList<Customer>();
//    private static const n = 1;
    public Customer getNewCustomer(){
        return this.customer;
    }

    public void setNewCustomer(Customer customer){
        this.customer = customer;
    }

    public static void main(String[] args){
        BarberShop bs = new BarberShop();
        Barber barber = new Barber();
        int n = Integer.valueOf(args[1]);
        while(true){
            if (!barber.isSleeping()){
                if(bs.chairs.size()>0){
                    barber.processCustomer(bs.chairs.poll());
                    if (bs.chairs.size()==0){
                        barber.setSleeping(true);
                    }
                }
            }
            if (bs.isNewCustomer){
                Customer c = bs.getNewCustomer();
                if (barber.isSleeping()){
                    barber.awake();
                }
                if (barber.getBusy()){
                    if (bs.chairs.size() < n){
                        bs.chairs.add(c);
                    }
                    else{
                        System.out.println("Customer goes to another Barber");
                    }
                } else{
                    barber.processCustomer(c);
                    if (bs.chairs.size()==0){
                        barber.setSleeping(true);
                    }
                }
            }
        }
    }
}
